### ISSUE TITLE
Overriding blocks

### ISSUE THREAD

**Issue by lnicola** [getzola/zola#361]
I'm not familiar to Jinja2 templates, so I'm probably missing something. The `hyde` theme defines a couple of blocks like `extra_head` and `sidebar_about`. Can I override them in my site? I tried dropping a file in either `templates/` or `themes/hyde/templates/`, but that didn't seem to work.

**Comment by Keats** [comment]
You can override blocks, you first need to `extends` from a template and then redefining them (see https://tera.netlify.com/docs/templates/#inheritance for the docs).

In short for example in the hyde template, you would create a `index.html` template in `templates/` that  looks like:

```jinja
{% extends "themes/hyde/index.html" %}

{% block extra_head %}
// You can put what you want there
{% endblock %}
```

**Comment by lnicola** [comment]
Got it, that's just like `page.html` extends `index.html`. Thanks.

**Comment by lnicola** [comment]
Hmm, that doesn't work:

```
Error: Template 'index.html' is inheriting from 'themes/hyde/templates/index.html', which doesn't exist or isn't loaded.
```

The file does exist, so maybe it's not loaded yet?

**Comment by Keats** [comment]
Sorry I got it wrong, it should be `{% extends "hyde/templates/index.html" %}`

**Comment by lnicola** [comment]
Thanks, that works.

**Comment by lnicola** [comment]
Actually, this doesn't seem great. I can make a template that extends the theme `index.html` template and override that block. However, the theme's `post.html` will the theme's own `index.html`, not mine.

**Comment by Keats** [comment]
As far as I can tell it is working as expected, the page template doesn't really know about your updated `index.html`.
The renderer does those lookups:

- for rendering index page: `templates/index.html` extends `hyde/templates/index.html`
- for rendering a post page: `hyde/templates/page.html` extends `hyde/templates/index.html`

The renderer doesn't look doesn't see `templates/index.html` at all in the extend path of the page template so it assumes it's done. 
It's not ideal I agree but I don't see an easy way for the renderer to understand it needs to insert `templates/index.html` in the middle: we would need to rewrite manually the inheritance chain AFAIK

**Comment by lnicola** [comment]
I realized it was working as designed, but it seems to limit the usefulness of being able to place a theme in `themes/`. Maybe the recommended approach (for now, at least) is to copy a theme and change it instead of using overrides?

**Comment by JeanMertz** [comment]
The issue I found with this is _somewhat_ similar, so I figured I'd post it here, but feel free to move this if it's a separate issue:

I've installed a theme, and want to modify its `index.html`. I can copy the file to `templates/` and make the modifications, but the index also references a `macro.html` that is included by the theme.

I don't intent to override that file, but it seems like I have no choice as Zola doesn't know that the theme already supplies this file:

```
Error: Template `index.html` loads macros from `macros.html` which isn't present in Tera
```

**Comment by southerntofu** [comment]
Does [this](https://github.com/southerntofu/zola/commit/5b618b3ff5b7bb86312d32c2530c04cbe764b118) solve your problems? I put in the commit a small test_site2/ folder so you can visualize how that works. EDIT: didn't test if it also fixes #324 by chance.

[This](https://tildegit.org/southerntofu/zola-water/commit/841b6d4cc477984b6f217da61a891be5f72b2a37) is the trick i used to make my theme work with taxonomies (with taxonomies templates in site folder transparently extending theme's index.html).

PR welcome?
